### PR TITLE
Fix custom operation_id's from OpenAPI

### PR DESCRIPTION
**Pull request by fao89**
https://pulp.plan.io/issues/7341
closes #7341

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/openapi/__init__.py [hunk 1]
```diff
@@ -65,20 +65,23 @@ def get_tags(self):
 
         return tags
 
+    def get_operation_id_action(self):
+        """Get action from operation_id."""
+        action_name = getattr(self.view, "action", self.method.lower())
+        if action_name not in ["retrieve", "list", "destroy", "create"]:
+            return action_name
```
Comment (fao89) [inline pulpcore/openapi/__init__.py:72]: I'm looking for customized action such as: `sync` and `modify`.
So if we get a customized action, we return it.

File: pulpcore/openapi/__init__.py [hunk 2]
```diff
@@ -65,20 +65,23 @@ def get_tags(self):
...
+
+        if self.method.lower() == "get" and self._is_list_view():
+            return "list"
+
+        return self.method_mapping[self.method.lower()]
```
Comment (fao89) [inline pulpcore/openapi/__init__.py:77]: the default actions are handled here

**Review by ipanova — APPROVED** [review]

File: pulpcore/openapi/__init__.py [hunk 3]
```diff
@@ -296,14 +299,13 @@ def parse(self, request, public):
...
-                if schema.method.lower() == "get" and schema._is_list_view():
-                    operation["operationId"] = "list"
-                elif action_name not in schema.method_mapping:
-                    action = action_name.replace("destroy", "delete").replace("retrieve", "read")
+                tokenized_path = schema._tokenize_path()
+                tokenized_path = "_".join(
+                    [t.replace("-", "_").replace("/", "_").lower() for t in tokenized_path]
+                )
+                action = schema.get_operation_id_action()
+                if f"{tokenized_path}_{action}" == operation["operationId"]:
```
Comment (fao89) [inline pulpcore/openapi/__init__.py:307]: this assures to return only the action for the bindings.
`rpm_rpm_repositories_list` -> `list`
`rpm_rpm_repositories_sync` - > `sync`

File: pulpcore/openapi/__init__.py [hunk 4]
```diff
@@ -296,14 +299,13 @@ def parse(self, request, public):
...
             # operationId as actions [list, read, sync, modify, create, delete, ...]
             if request and "bindings" in request.query_params:
-                action_name = getattr(view, "action", schema.method.lower())
-                if schema.method.lower() == "get" and schema._is_list_view():
-                    operation["operationId"] = "list"
-                elif action_name not in schema.method_mapping:
-                    action = action_name.replace("destroy", "delete").replace("retrieve", "read")
+                tokenized_path = schema._tokenize_path()
+                tokenized_path = "_".join(
+                    [t.replace("-", "_").replace("/", "_").lower() for t in tokenized_path]
```
Comment (dralley) [inline pulpcore/openapi/__init__.py:304]: Could you explain what's going on here?

Comment (fao89) [reply pulpcore/openapi/__init__.py:354]: Please check if the docstrings are enough or if I have to make an inline comment here

**Pull request merged**